### PR TITLE
Converts purge functions into a class

### PR DESCRIPTION
**Pull request by pattonwebz**
Moves the purge methods into a class and sets them as static since they don't require any instantiation or setup.

The only functional changes I made were:
* adding a condition to check if the post type exists when operating on CPTs to save needing a query to run if no CPT exists that would match against it
* adding some type hints on the methods.
* removed some single use variables in the methods
* fixed issue tests discovered that's outlined in #516 

Closes: #463 
Fixes: #516

### CODE REVIEW

File: admin/class-purge-post-data.php [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	/**
+	 * Purge deleted posts
+	 *
+	 * @param int $post_id ID of the post.
+	 *
+	 * @return void
+	 */
+	public static function delete_post( int $post_id ) {
+		global $wpdb;
+		$site_id    = get_current_blog_id();
```
Comment (SteveJonesDev) [inline admin/class-purge-post-data.php:24]: I know this is the way it was before but we can avoid these single-use variables in all these methods. We can also improve readability by using multiple lines of code.

Example:
```php
    $wpdb->query( 
        $wpdb->prepare( 
            "DELETE FROM %i WHERE postid = %d AND siteid = %d", 
            $wpdb->prefix . 'accessibility_checker', 
            $post_id, 
            get_current_blog_id() 
        )
    );
```

**Review by SteveJonesDev — CHANGES_REQUESTED** [review]
@pattonwebz, I added some comments. I think @aristath is going to review this as well, so he might have some feedback too.

File: tests/phpunit/PurgePostDataTest.php [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+ * @package Accessibility_Checker
+ */
+
+use PHPUnit\Framework\TestCase;
+use EDAC\Admin\Purge_Post_Data;
+
+/**
+ * Class PurgePostDataTest
+ */
+class PurgePostDataTest extends TestCase {
```
[the remark this answers is not in the document]
Comment (pattonwebz) [reply tests/phpunit/PurgePostDataTest.php:14]: I didn't use any methods that don't exist on the base `TestCase` class, which was the only reason. For consistency with other tests, I can swap it to `WP_UnitTestCase` for consistency and refactor it a little.

File: tests/phpunit/PurgePostDataTest.php [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+		$this->assertEmpty( get_post_meta( $this->valid_post_id, '_edac', true ) );
+		$this->assertEmpty( get_post_meta( $this->valid_post_id, '_edacp', true ) );
+		$this->assertNotEmpty( get_post_meta( $this->valid_post_id, 'other', true ) );
+	}
+
+	/**
+	 * Test that Purge_Post_Data::delete_post() is called when the wp_trash_post action fires.
+	 */
+	public function testDeletePostIsCalledOnWpTrashPost() {
```
[the remark this answers is not in the document]
Comment (pattonwebz) [reply tests/phpunit/PurgePostDataTest.php:87]: I added a test for that method, and it also found an issue with the code under test. A mismatch on upper vs lower case table names was causing the data not to get deleted in the custom table.

**Comment by pattonwebz** [comment]
> @pattonwebz, I'm approving this PR but I did run into a bug while testing, but I don't think it's related. Can you confirm that's the case before merging this PR?
> 
> #535

I was able to reproduce that issue while working on a different branch without these changes so #535 seems unrelated to the changes here and I'll merge this shortly. Posting more details about what I found about that error on the issue thread when I have a minute outline it.

**Pull request merged**